engine/storage/datamotion/src/main/java/org/apache/cloudstack/storage/motion/StorageSystemDataMotionStrategy.java

```diff
@@ -1796,31 +1889,20 @@ private void createVolumeFromSnapshot(SnapshotInfo snapshotInfo) {
      * invocation of createVolumeFromSnapshot(SnapshotInfo).
      */
     private void deleteVolumeFromSnapshot(SnapshotInfo snapshotInfo) {
-        VolumeVO volumeVO = null;
-        // cleanup any temporary volume previously created for copy from a snapshot
-        if ("true".equalsIgnoreCase(snapshotInfo.getDataStore().getDriver().getCapabilities().get("CAN_CREATE_TEMP_VOLUME_FROM_SNAPSHOT"))) {
-            SnapshotDetailsVO tempUuid = null;
-            tempUuid = _snapshotDetailsDao.findDetail(snapshotInfo.getId(), "TemporaryVolumeCopyUUID");
```

Comment:
is there covered any where in your new code to delete the snapshot detail ?

Reply:
Yes this is part of the previous comment about changing how the temporary volume logic worked in the drivers.  This logic is all handled directly by the Adaptive provider code now.